### PR TITLE
Features/tag

### LINKED ISSUE #29: Allergy and preference
### Issuer:
Hung Ngoos
### Requirements:
- [ ] please list what tasks need to be done (be specific here)
- Create a list of tags and title 

### Dependencies:
- Please list the files that might be involved in this issue

### Done requirements:
- When will this issue count as done?

### PR DESCRIPTION
**Pull request by viennguyen24**
**Issue: #29 Allergy and Preferences**
**Branch: features/tag**
**PR: Create a custom tag component to display allergy and diet preferences** 

**Demo:**
[image](https://drive.google.com/file/d/1KYksRr_-2mwxhhfaZVpAfO9Ycjwa3yhF/view?usp=sharing)

**Code:**
```typescript
export type Allergy = "Dairy" | "Egg" | "Gluten" | "Grain" | "Peanut" | "Seafood" | "Sesame" | "Shellfish" | "Soy" | "Sulfite" | "Tree Nut" | "Wheat";
export type Diet = "Gluten Free" | "Ketogenic" | "Vegetarian" | "Lacto-Vegetarian" | "Ovo-Vegetarian" | "Vegan" | "Pescetarian" | "Paleo" | "Primal" | "Whole30" | "Low FODMAP"
export const allergyTypes: Allergy[] = ["Dairy", "Egg", "Gluten", "Grain", "Peanut", "Seafood", "Sesame", "Shellfish", "Soy", "Sulfite", "Tree Nut", "Wheat"];
export const dietTypes: Diet[] = ["Gluten Free", "Ketogenic", "Vegetarian", "Lacto-Vegetarian", "Ovo-Vegetarian", "Vegan", "Pescetarian", "Paleo", "Primal", "Whole30", "Low FODMAP"];

<HStack flexWrap="wrap" marginBlock={4}>
      {allergyTypes.map((type) => (
        <CustomTag key={type} type={type} canClose={true} />
      ))}
</HStack>
<HStack flexWrap="wrap">
      {dietTypes.map((type) => (
        <CustomTag key={type} type={type} canClose={true} />
      ))}
</HStack>
```
**Note:** You need to pass in the type prop with the exact string for allergy / diet

### CODE REVIEW

File: client/src/components/meals/tag.tsx [hunk 1]
```diff
@@ -0,0 +1,80 @@
...
+        background: 'red.300',
+        text: 'gray.100',
+        icon: MdNoMeals,
+      },
+      diet: {
+        background: 'green.400',
+        text: 'gray.100',
+        icon: MdOutlineBrunchDining,
+      },
+    };
```
Comment (hungngodev) [inline client/src/components/meals/tag.tsx]: This is a good scheme

**Review by hungngodev — CHANGES_REQUESTED** [review]
Could you upload the latest UI? I think you have not posted the latest one. I'm only seeing the tag component here.

**Comment by viennguyen24** [comment]
> Could you upload the latest UI? I think you have not posted the latest one. I'm only seeing the tag component here.

Hi Hung, I just uploaded the new UI with some sample code!

File: client/src/components/meals/tag.tsx
Comment (viennguyen24) [inline client/src/components/meals/tag.tsx:1]: This file is an old version of the CustomTag but at the moment I cant find a way to remove this

File: client/src/components/meals/CustomTag.tsx [hunk 2]
```diff
@@ -0,0 +1,105 @@
...
+    const [isOpen, setOpen] = useState<boolean>(true);
+    const defaultClose = () =>  {setOpen(!isOpen); console.log("clicked")};
+    return (
+      <>
+        {isOpen && (
+            <Tag
+            {...props}
+            ref={ref}
+            onClick={handleClick}
+            boxShadow="md"
```
Comment (hungngodev) [inline client/src/components/meals/CustomTag.tsx]: create a form for this!

**Review by hungngodev — CHANGES_REQUESTED** [review]
Please merge the conflict!

**Review by hungngodev — CHANGES_REQUESTED** [review]
Please merge any remaining conflicts

**Comment by viennguyen24** [comment]
> Please merge any remaining conflicts

Hi Hung I just resolved all conflicts at [#76e2a42](https://github.com/hungngodev/Flavorie/pull/44/commits/76e2a42b1692c25a43860073cc7d00e66711b4dc)

**Comment by hoatuyet423** [comment]
- conflicts not fully resolved still

**Review by hungngodev — APPROVED** [review]
Merging conflict done. Thankss

**Pull request merged**